internal.sh

```diff
@@ -0,0 +1,89 @@
+#    - As of now this script works with 4.11 releases because only `rhaos-4.11-rhel-8` branch
+#      created in dist-git and tested.
+# Usage:
+# If you want to build latest candidate stream for 4.11
+#   - ./internal.sh
+# If you want to build specific version of 4.11.3
+#   - OPENSHIFT_VERSION=4.11.3 ./internal.sh
+
+set -exuo pipefail
+
+export LC_ALL=C.UTF-8
+export LANG=C.UTF-8
+
+# kill all the child processes for this script when it exits
+trap 'jobs=($(jobs -p)); [ -n "${jobs-}" ] && ((${#jobs})) && kill "${jobs[@]}" || true' EXIT
+
+rm -fr crc-cluster-kube-apiserver-operator
+rm -fr crc-cluster-kube-controller-manager-operator
+
+function run_preflight_checks() {
+        if [ -z "${OPENSHIFT_PULL_SECRET_PATH-}" ]; then
+            echo "OpenShift pull secret file path must be specified through the OPENSHIFT_PULL_SECRET_PATH environment variable"
+            exit 1
+        elif [ ! -f ${OPENSHIFT_PULL_SECRET_PATH} ]; then
+            echo "Provided OPENSHIFT_PULL_SECRET_PATH (${OPENSHIFT_PULL_SECRET_PATH}) does not exists"
+            exit 1
+        fi
+}
+
+run_preflight_checks
+
+MIRROR=${MIRROR:-https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/ocp}
+
+# If user defined the OPENSHIFT_VERSION environment variable then use it.
+if test -n "${OPENSHIFT_VERSION-}"; then
+    OPENSHIFT_RELEASE_VERSION=${OPENSHIFT_VERSION}
+    echo "Using release ${OPENSHIFT_RELEASE_VERSION} from OPENSHIFT_VERSION"
+else
+    OPENSHIFT_RELEASE_VERSION="$(curl -L "${MIRROR}"/candidate-4.11/release.txt | sed -n 's/^ *Version: *//p')"
+    if test -n "${OPENSHIFT_RELEASE_VERSION}"; then
+        echo "Using release ${OPENSHIFT_RELEASE_VERSION} from the mirror"
+    else
+        echo "Unable to determine an OpenShift release version.  You may want to set the OPENSHIFT_VERSION environment variable explicitly."
+        exit 1
+    fi
+fi
+
+if test -z "${OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE-}"; then
+
+    OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE="$(curl -L "${MIRROR}/${OPENSHIFT_RELEASE_VERSION}/release.txt" | sed -n 's/^Pull From: //p')"
+elif test -n "${OPENSHIFT_VERSION-}"; then
+    echo "Both OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE and OPENSHIFT_VERSION are set, OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE will take precedence"
+    echo "OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE: $OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE"
+    echo "OPENSHIFT_VERSION: $OPENSHIFT_VERSION"
+fi
+echo "Setting OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE to ${OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE}"
+
+mkdir -p openshift-clients/linux
+curl -L "${MIRROR}/${OPENSHIFT_RELEASE_VERSION}/openshift-client-linux-${OPENSHIFT_RELEASE_VERSION}.tar.gz" | tar -zx -C openshift-clients/linux oc
+OC=./openshift-clients/linux/oc
+
+function patch_and_push_image() {
+    local image_name=$1
+    openshift_version=$(${OC} adm release info -a ${OPENSHIFT_PULL_SECRET_PATH} ${OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE} -ojsonpath='{.config.config.Labels.io\.openshift\.release}')
```

Comment:
In case override image variable in used we should use this way to determine the version.

Reply:
It's minor, but this would be more obvious if done only when `OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE` is set

Reply:
@cfergeau it would be then 2 different codepath, better to have it single for both usecase instead of determining the `version` separately in case image override or not.